src/traces/splom/index.js

```diff
@@ -59,16 +59,17 @@ function calc(gd, trace) {
             xa = AxisIDs.getFromId(gd, trace._diag[i][0]);
             ya = AxisIDs.getFromId(gd, trace._diag[i][1]);
 
+            // if corresponding x & y axes don't have matching types, skip dim
+            if(xa && ya && xa.type !== ya.type) continue;
```

Comment:
Seems likely to confuse people if they encounter this issue... do you want to include a `Log` message, or perhaps find a way to surface this in `Plotly.validate`?

Reply:
As this is part of the calc step (which is outside the scope of `Plotly.validate`) and that we can't easily move this to the defaults step (as axis defaults are coerced after trace defaults), I chose to only log something in https://github.com/plotly/plotly.js/pull/2899/commits/182390442ee0843e022e32a311baf138a18c4570

Note that I chose I was first planning on using `Lib.warn` instead of `Lib.log`, but currently the image server bails whenever a mock results in a `Lib.warn`, so the mock added in https://github.com/plotly/plotly.js/pull/2899/commits/390f29202e4230035e094e9731f4f643e5ca9ea9 would fail to generate an image. I hope you're ok with just a `Lib.log` here.